src/plugins/robot-memory/event_trigger_manager.cpp

```diff
@@ -124,16 +129,16 @@ EventTriggerManager::remove_trigger(EventTrigger *trigger)
 	delete trigger;
 }
 
-QResCursor
-EventTriggerManager::create_oplog_cursor(mongo::DBClientBase *con,
-                                         std::string          oplog,
-                                         mongo::Query         query)
+cursor
+EventTriggerManager::create_oplog_cursor(mongocxx::collection &coll, bsoncxx::document::view query)
 {
-	QResCursor res = con->query(oplog, query, 0, 0, 0, QueryOption_CursorTailable);
-	//Go to end of Oplog to get new updates from then on
-	while (res->more()) {
-		res->next();
-	}
+	mongocxx::options::find opts;
+	opts.cursor_type(mongocxx::cursor::type::k_tailable_await);
+	// TODO max time/timeouts?
+	cursor res = coll.find(query, opts);
+	// Go to end of oplog to get new updates from then on.
+	// Note that res->begin() forwards the cursor, no need to call next() or something similar.
+	while (res.begin() != res.end()) {}
```

Comment:
https://stackoverflow.com/a/35856472/4738953 may provider a nice version, getting the collection size and then using `std::next` to advance the cursor thus far. However, if the oplog still increases while doing that, you end up not being at the end. On the other hand, the given code may not terminate at all in that case.

Just saying that it may be worth thinking about this a bit more, and possible having an exit condition in the loop to avoid at least the non-termination.

Reply:
Non-termination could only occur if we get new changes continuously and reading the change stream is slower than those changes occur. I don't think that's a realistic problem. I made progressing the change stream slightly more explicit in b5d22d1b111e06c385bf3ffcfc9091e2d94906ae though.

Reply:
I plan to use [timestamps](http://mongocxx.org/api/current/classmongocxx_1_1options_1_1change__stream.html#a8908bd0fd0c5d76962b5b4d07c465a08) and [resume documents](http://mongocxx.org/api/current/classmongocxx_1_1options_1_1change__stream.html#a3b37cbcca59d0ab2d834396e6a8c0bff) for creating triggers so we don't need to forward the cursor at all anymore. However, that's not fully working yet and should probably not be part of this PR.